### DISCUSSION TITLE
503 error in docker scrypted, works fine in home assistant

### DISCUSSION THREAD

**Discussion by Phil-Barker** [koush/scrypted#823] [Q&A] [answered]
I don't know if anyone can help with this but I'm beating my head against a wall

I have scrypted running fine within HomeAssistant - 4 TP-Link Kasas cameras that work in a hacky way of having to sender basic HTTP authentication to them

I want to migrate to running scrypted in Docker on my Unraid server instead because I want to use nvidia gpu for transcoding.

Old scrypted within HomeAssistant - 
```
[Rebroadcast Plugin]: Stream 1 prebuffer session started
[Rebroadcast Plugin]: prebuffer session starting
[Rebroadcast Plugin]: Selected stream Stream 1
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -headers Authorization: Basic XXX -f h264 -i https:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:44203/d5d12c7c7f85dfb3
[Rebroadcast Plugin]: [https @ 0x6128300] No trailing CRLF found in HTTP header. Adding it.
[Rebroadcast Plugin]: Input #0, h264, from 'https://192.168.1.207:19443/https/stream/mixed?video=h264&audio=g711&resolution=hd':
[Rebroadcast Plugin]:   Duration: N/A, bitrate: N/A
[Rebroadcast Plugin]:   Stream #0:0: Video: h264 (Main), yuvj420p(pc, bt709, progressive), 1920x1080, 25 fps, 25 tbr, 1200k tbn
[Rebroadcast Plugin]: No audio stream detected.
[Rebroadcast Plugin]: Video codec is not h264. If there are errors, try changing your camera's encoder output.
[Rebroadcast Plugin]: prebuffer session started
[Rebroadcast Plugin]: sending prebuffer 4000
[Rebroadcast Plugin]: Stream 1 active rebroadcast clients: 1
```

New scrypted within docker - 
```
[Rebroadcast Plugin]: prebuffer sessions starting in 5 seconds
[Rebroadcast Plugin]: Stream 1 prebuffer session started
[Rebroadcast Plugin]: prebuffer session starting
[Rebroadcast Plugin]: -hide_banner -fflags +genpts -headers Authorization: Basic XXX -f h264 -i https:[REDACTED] -rtsp_transport tcp -vcodec copy -acodec copy -f rtsp rtsp://127.0.0.1:36631/90b01a580b1ec400
[Rebroadcast Plugin]: [https @ 0x58aa300] No trailing CRLF found in HTTP header. Adding it.
[Rebroadcast Plugin]: [https @ 0x58aa300] HTTP error 503 Service Unavailable
[Rebroadcast Plugin]: https://192.168.1.207:19443/https/stream/mixed?video=h264&audio=g711&resolution=hd: Server returned 5XX Server Error reply
[Rebroadcast Plugin]: ffmpeg exited
[Rebroadcast Plugin]: rebroadcast error Error: ffmpeg exited
```

Docker container is running on host networking and appears to connect fine. If I change any of the basic auth I get a 401 error instead so it definitely seems to be talking to the camera but getting a 503 🤷 

The scrypted version is the same, ffmpeg plugin version is the same. There's no difference at all between the two yet the docker version says it is getting a 503 error.

Has anyone at all got any insight into what might be causing this?

**Comment by Phil-Barker** [comment, marked as answer]
Firstly my apologies for raising this as an issue instead of a discussion.

I have resolved this now.
For anyone with the same problem in the future - it was simple. It was returning a 503 error because I still had scrypted running in home assistant AND the new one in docker - the camera was erroring as too many clients were trying to connect to the stream.

Face-palm moment.

**Comment by oOblik** [comment]
@Phil-Barker Did you manage to actually get Nvidia transcoding to work? 
I am also looking to do so, and hopefully also use it for tensorflow object detection but while my GPU shows up in nvidia-smi it does not seem to be detected by plugins.
